### DISCUSSION TITLE
Running part of the code on the web

### DISCUSSION THREAD

**Discussion by bonninr** [GrandOrgue/OdfEdit#62] [General]
Hello,
I've been intending to produce some web based organ edition utilities, and found that the excellent Python classes from ODFEdit are all I needed to understand and represent those files. So I extracted the main data classes, and was able to run them in a browser, loading a sample Organ File, and accessing the data dictionary from a web browser, using the Pyodide framework. There's no more functionality than that at the moment, but I'll keep you posted if it's of interest.
Best regards.
![image](https://github.com/user-attachments/assets/e2aea039-4cc6-40f9-8f0a-d44e37c31384)
![image](https://github.com/user-attachments/assets/49fa44e9-f618-4706-a228-ff40bd248138)

**Comment by eturpault** [comment]
Interresting reuse of the source code of OdfEdit 👍

**Comment by bonninr** [comment]
Hi, I wanted to let you know I've built a web-based frontend for OdfEdit that runs entirely in the browser: **OdfEdit Web**, available at https://www.openpipes.org/odfweb/

It works by running the original OdfEdit Python script unchanged inside the browser using **Pyodide** (Python compiled to WebAssembly), with a React interface around it. No installation required — the Python logic runs locally in the user's tab.

It supports opening and editing `.organ` files, inline attribute editing, ODF validation, and the Hauptwerk → GrandOrgue conversion functionality.

All the core logic is this project's original work — I'm simply wrapping it to make it accessible without requiring a Python installation. A link to this repository is included in the app.

I wanted to share it here in case it's useful to the community. Happy to answer any questions.

**Reply by eturpault** [reply]
Thank you for giving to users this way to use OdfEdit without any installation. I have added this option in the README of the project.

**Reply by eturpault** [reply]
@bonninr : it would be good to display somewhere in the web page which version of OdfEdit is used.
